mono/eglib/gdate-unix.c

```diff
@@ -46,11 +46,26 @@ g_get_current_time (GTimeVal *result)
 void
 g_usleep (gulong microseconds)
 {
-	struct timespec req, rem;
+	struct timespec req;
 
 	req.tv_sec = microseconds / 1000000;
 	req.tv_nsec = (microseconds % 1000000) * 1000;
-	
+
+#ifdef HAVE_CLOCK_NANOSLEEP
+	int ret;
+
+	// Use clock_nanosleep if available to prevent time drifting problems when
+	// repeatedly signaling the thread for suspend during sleep.
+	do {
+		ret = clock_nanosleep (CLOCK_MONOTONIC, TIMER_ABSTIME, &req, NULL);
```

Comment:
mono-threads.c has similar code but disables it on Android for some reason: https://github.com/mono/mono/blob/08feb423c1f214a5341fcb0132054cd0dcf9f891/mono/utils/mono-threads.c#L1677-L1695

Reply:
I'd assume that code precedes the HAVE_CLOCK_NANOSLEEP define

Reply:
The android check was introduced by https://github.com/mono/mono/commit/c40af3e9a08cd9c9e905d299192acab3236e6d0d. Let's hope the configure check actually works against the NDK, I know we had issues with it in the past.